pyravendb/raven_operations/timeseries_operations.py

```diff
@@ -0,0 +1,161 @@
+        self.appends.append(TimeSeriesOperation.AppendOperation(timestamp, values, tag))
+
+    def remove(self, from_date: datetime, to_date: datetime):
+        self.removals.append(TimeSeriesOperation.RemoveOperation(from_date, to_date))
+
+    def to_json(self):
+        if self.appends and isinstance(self.appends, Generator):
+            self.appends = next(self.appends)
+        return {"Name": self.name, "Appends": [a.to_json() for a in self.appends] if self.appends else self.appends,
+                "Removals": [r.to_json() for r in self.removals] if self.removals else self.removals}
+
+    class AppendOperation:
+        def __init__(self, timestamp: datetime, values: List[float] or float, tag: Optional[str] = None):
+            if not timestamp:
+                raise ValueError("Missing timestamp property")
+            self.timestamp = timestamp
+            if values is None:
+                raise ValueError("Missing values property")
+            try:
+                if not isinstance(values, list):
+                    values = [float(values)]
+                else:
+                    values = [float(value) for value in values]
+            except ValueError:
+                raise (f"One or more values from values are not numbers %s" % values)
+            self.values = values
+            self.tag = tag
+
+        def to_json(self):
+            _dict = {"Timestamp": Utils.datetime_to_string(self.timestamp), "Values": self.values}
+            if self.tag:
+                _dict["Tag"] = self.tag
+            return _dict
+
+    class RemoveOperation:
+        def __init__(self, from_date: datetime, to_date: datetime):
+            if not from_date:
+                raise ValueError("Missing from_date property")
+            if not to_date:
+                raise ValueError("Missing to_date property")
+
+            self.from_date = from_date
+            self.to_date = to_date
+
+        def to_json(self):
+            return {"From": self.from_date, "To": self.to_date}
+
+
+class TimeSeriesBatchOperation(Operation):
+    def __init__(self, document_id: str, operation: TimeSeriesOperation):
+
+        super().__init__()
+        self._document_id = document_id
+        self.__operation = operation
+
+    def get_command(self, store, conventions, cache=None):
+        return self._TimeSeriesBatchCommand(self._document_id, self.__operation)
+
+    class _TimeSeriesBatchCommand(RavenCommand):
+        def __init__(self, document_id: str, operation: TimeSeriesOperation):
+            super().__init__(method="POST")
+
+            if not document_id:
+                raise ValueError("Invalid document_id")
+
+            self._document_id = document_id
+            self._operation = operation
+
+        def create_request(self, server_node):
+            self.url = (f"{server_node.url}/databases/{server_node.database}"
+                        f"/timeseries?id={Utils.quote_key(self._document_id)}")
```

Comment:
Does this work? I thought we need to send a `DocumentId` field in the JSON

Reply:
Yes I tested everything, this is exactly like the c# client, we send the document id in the URL